### PR TITLE
Issue 331/mention support on @ keypress

### PR DESCRIPTION
**Pull request by SergioEstevao**
## Description

Change the rich text component in mobile in order to intercept any random keypress.

## How has this been tested?

This can be tested by running: `npm run native ios` or `npm run native android` after the demo app starts, follow these steps:

 - Add a Paragraph block
 - Tap the @ on the keyboard
 - Check if a mention to `matt` is added
 - Check if the mention is only inserted if the @ is inserted with a space before or at the start of the paragraph.
 - Check if not mention is inserted on the title of the post.

Related PR: GB-Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2424

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/rich-text/src/component/index.native.js [hunk 1]
```diff
@@ -294,7 +298,20 @@ export class RichText extends Component {
...
 	}
 
+	onKeyDown( event ) {
+		if ( event.defaultPrevented ) {
+			return;
+		}
+
+		this.handleDelete( event );
+		this.handleEnter( event );
+		this.handleMention( event );
```
Comment (mchowning) [inline packages/rich-text/src/component/index.native.js:308]: Just a reminder that before merging this we need to either add a `DEV` flag here or remove it from where we add the mention button to the format bar.

**Comment by SergioEstevao** [comment]
@ceyhun do you mind giving a look to this again, now that we have updated it for all the monorepo changes?

**Review by ceyhun — APPROVED** [review]
It works great and as expected, although on Android it felt a bit slow to react when @ is pressed (1-2 seconds) for me both in emulator and in real-device. 
It may be because it was running from the server and in DEV mode or because I have an old device. It'll probably be faster when bundled.

File: packages/react-native-aztec/src/AztecView.js [hunk 2]
```diff
@@ -57,21 +59,43 @@ class AztecView extends React.Component {
...
-		if ( ! this.props.onEnter ) {
+		if ( ! this.props.onKeyDown ) {
 			return;
 		}
 
-		const { onEnter } = this.props;
-		onEnter( event );
+		const { onKeyDown } = this.props;
+
+		const newEvent = { ...event, keyCode: 13 };
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply packages/react-native-aztec/src/AztecView.js]: Good suggestion, I will update to use the keycodes package.

**Pull request merged**